plugins/animate/squeezimize.hpp

```diff
@@ -0,0 +1,388 @@
+            self->animation_geometry.height =
+                std::max(std::max(std::max(src_box.height,
+                    self->minimize_target.height),
+                    (self->minimize_target.y + self->minimize_target.height) - src_box.y),
+                    (src_box.y + src_box.height) - self->minimize_target.y);
+
+            const float vertex_data_pos[] = {
+                1.0f * self->animation_geometry.x,
+                1.0f * self->animation_geometry.y + self->animation_geometry.height,
+                1.0f * self->animation_geometry.x + self->animation_geometry.width,
+                1.0f * self->animation_geometry.y + self->animation_geometry.height,
+                1.0f * self->animation_geometry.x + self->animation_geometry.width,
+                1.0f * self->animation_geometry.y,
+                1.0f * self->animation_geometry.x, 1.0f * self->animation_geometry.y,
+            };
+
+            const glm::vec4 src_box_pos{
+                float(src_box.x - self->animation_geometry.x) / self->animation_geometry.width,
+                float(src_box.y - self->animation_geometry.y) / self->animation_geometry.height,
+                float((src_box.x - self->animation_geometry.x) + src_box.width) /
+                self->animation_geometry.width,
+                float((src_box.y - self->animation_geometry.y) + src_box.height) /
+                self->animation_geometry.height
+            };
+
+            const glm::vec4 target_box_pos{
+                float(self->minimize_target.x - self->animation_geometry.x) / self->animation_geometry.width,
+                float(self->minimize_target.y - self->animation_geometry.y) / self->animation_geometry.height,
+                float((self->minimize_target.x - self->animation_geometry.x) + self->minimize_target.width) /
+                self->animation_geometry.width,
+                float((self->minimize_target.y - self->animation_geometry.y) + self->minimize_target.height) /
+                self->animation_geometry.height
+            };
+
+            OpenGL::render_begin(target);
+            self->program.use(wf::TEXTURE_TYPE_RGBA);
+            self->program.uniformMatrix4f("matrix", target.get_orthographic_projection());
+            self->program.attrib_pointer("position", 2, 0, vertex_data_pos);
+            self->program.attrib_pointer("uv_in", 2, 0, vertex_data_uv);
+            self->program.uniform1i("upward", upward);
+            self->program.uniform1f("progress", progress);
+            self->program.uniform4f("src_box", src_box_pos);
+            self->program.uniform4f("target_box", target_box_pos);
+            self->program.set_active_texture(src_tex);
+            GL_CALL(glDrawArrays(GL_TRIANGLE_FAN, 0, 4));
+            OpenGL::render_end();
+        }
+    };
+
+    squeezimize_transformer(wayfire_view view,
+        wf::geometry_t minimize_target, wf::geometry_t bbox) : wf::scene::view_2d_transformer_t(view)
+    {
+        this->view = view;
```

Comment:
I believe with the other suggested changes, this->view will not be needed.

Reply:
Indeed.

Reply:
Actually it is needed, for in gen_render_instances() when creating the simple_node_render_instance_t object instance.

Reply:
On second take, I managed to not need to store it.